File: WeatherNotes/weather.js
// Weather API configuration
const API_KEY = process.env.OPENWEATHER_API_KEY || '';

async function getWeatherData() {
    try {
        const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=Mexico City,MX&appid=${API_KEY}&units=metric&lang=es`);
        const data = await response.json();
        
        if (response.ok) {
            return {
                success: true,
                city: 'Ciudad de México',
                temperature: Math.round(data.main.temp),
                description: data.weather[0].description,
                feelsLike: Math.round(data.main.feels_like),
                humidity: data.main.humidity,
                windSpeed: Math.round(data.wind.speed * 3.6)
            };
        } else {
            return { success: false, error: 'API error' };
        }
    } catch (error) {
        return { success: false, error: 'Network error' };
    }
}

// Export for use in HTML
if (typeof module !== 'undefined' && module.exports) {
    module.exports = { getWeatherData };
}

File: WeatherNotes/lib/models/weather.dart
class Weather {
  final String cityName;
  final double temperature;
  final String description;
  final String main;
  final double feelsLike;
  final int humidity;
  final double windSpeed;
  final String iconCode;
  final DateTime timestamp;

  Weather({
    required this.cityName,
    required this.temperature,
    required this.description,
    required this.main,
    required this.feelsLike,
    required this.humidity,
    required this.windSpeed,
    required this.iconCode,
    required this.timestamp,
  });

  factory Weather.fromJson(Map<String, dynamic> json, String cityName) {
    return Weather(
      cityName: cityName,
      temperature: (json['main']['temp'] as num).toDouble(),
      description: json['weather'][0]['description'],
      main: json['weather'][0]['main'],
      feelsLike: (json['main']['feels_like'] as num).toDouble(),
      humidity: json['main']['humidity'],
      windSpeed: (json['wind']['speed'] as num).toDouble(),
      iconCode: json['weather'][0]['icon'],
      timestamp: DateTime.now(),
    );
  }

  String get iconUrl => 'https://openweathermap.org/img/wn/$iconCode@2x.png';

  String get temperatureString => '${temperature.round()}°C';
  
  String get feelsLikeString => '${feelsLike.round()}°C';

  @override
  String toString() {
    return 'Weather{cityName: $cityName, temperature: $temperature, description: $description}';
  }
}
